File: src/core/templates.rs
use std::collections::HashMap;

use crate::core::manifest::{Manifest, Package, Toolchain};


pub fn crate_manifest(name: &str) -> Manifest {
    use crate::core::manifest::Features;
    
    Manifest {
        package: Package {
            name: name.to_string(),
            version: "0.1.0".to_string(),
            edition: "cpp17".to_string(),
        },
        dependencies: HashMap::new(),
        toolchain: Toolchain::default(),
        features: Features::default(),
    }
}

pub fn get_main_template() -> &'static str {
    r#"#include <iostream>

int main() {
    std::cout << "Hello, world!" << std::endl;
    return 0;
}"#
}

pub fn get_gitignore_template() -> &'static str {
    r#"/target"#
}

File: src/core/templates.cpp
#include "core/templates.hpp"

namespace core::templates {

auto main_cpp_template() -> std::string {
    return R"(#include <print>

auto main() -> int {
    std::println("Hello, world!");
    return 0;
}
)";
}

auto gitignore_template() -> std::string { return "/target\n/packages\n"; }

}  // namespace core::templates
